infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -0,0 +1,362 @@
+        Set<String> sinkClustersUnchanged = Sets.intersection(newSinks, currentSinks);
+
+        topology = newTopology;
+        metadataManager.setTopologyConfigId(topology.getTopologyConfigId());
+
+        Set<LogReplicationSession> sessionsToRemove = new HashSet<>();
+        Set<LogReplicationSession> sessionsUnchanged = new HashSet<>();
+
+        try {
+            IRetry.build(IntervalRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(LogReplicationMetadataManager.NAMESPACE)) {
+                    sessions.forEach(session -> {
+                        if (sinksToRemove.contains(session.getSinkClusterId()) ||
+                                sourcesToRemove.contains(session.getSourceClusterId())) {
+                            sessionsToRemove.add(session);
+                            metadataManager.removeSession(txn, session);
+                        }
+
+                        if(sinkClustersUnchanged.contains(session.getSinkClusterId())) {
+                            sessionsUnchanged.add(session);
+                            sessionToContextMap.get(session).setTopologyConfigId(topology.getTopologyConfigId());
+                            metadataManager.updateReplicationMetadataField(txn, session,
+                                    ReplicationMetadata.TOPOLOGYCONFIGID_FIELD_NUMBER, topology.getTopologyConfigId());
+                        }
+                    });
+                    txn.commit();
+                } catch (TransactionAbortedException e) {
+                    throw new RetryNeededException();
+                }
+
+                stopReplication(sessionsToRemove, true);
+                updateReplicationParameters(sessionsUnchanged);
+                createSessions();
+                return null;
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Unrecoverable exception while removing session", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    private void updateReplicationParameters(Set<LogReplicationSession> sessionsUnchanged) {
+        if (replicationManager == null) {
+            return;
+        }
+        for (LogReplicationSession session : sessionsUnchanged) {
+            ClusterDescriptor cluster = topology.getSinkClusters().get(session.getSinkClusterId());
+            replicationManager.refreshRuntime(session, cluster, topology.getTopologyConfigId());
+        }
+    }
+
+    /**
+     * Create sessions (outgoing and incoming) along with metadata
+     *
+     */
+    private void createSessions() {
+        LogReplicationSession session;
+
+        try(TxnContext txn = corfuStore.txn(LogReplicationMetadataManager.NAMESPACE)) {
+            // TODO(V2): A replication context should be specific per session, as the set of streams to replicate is
+            // specific to a session (for now only supporting one default model) so using the same context
```

Comment:
yes I too have the same idea.  Infact, we currently have a map of subscriber -> streams in the config manager(in lr_v2).  I think this PR removed it because that part is not fully defined yet

Reply:
removing the TODO based on this discussion.